app/controllers/project/Versions.scala

```diff
@@ -527,25 +517,40 @@ class Versions @Inject()(stats: StatTracker,
             versionId = version.id.get,
             address = InetString(StatTracker.remoteAddress)))
 
-          if (wget) {
-            Future.successful(
-            MultipleChoices(this.messagesApi("version.download.confirm.wget"))
-              .withHeaders("Content-Disposition" -> "inline; filename=\"README.txt\""))
-          } else if (curl) {
-            Future.successful(MultipleChoices(this.messagesApi("version.download.confirm.body.plain",
-              self.confirmDownload(author, slug, target, Some(dlType.id), token).absoluteURL(),
-              CSRF.getToken.get.value) + "\n")
-              .withHeaders("Content-Disposition" -> "inline; filename=\"README.txt\""))
-          } else if (api.getOrElse(false)) {
-            Future.successful(MultipleChoices(Json.obj(
-              "message" -> this.messagesApi("version.download.confirm.body.api").split('\n'),
-              "post" -> helper.CSRF(
-                self.confirmDownload(author, slug, target, Some(dlType.id), token)).absoluteURL())))
+          if (api.getOrElse(false)) {
+            warning.map(warn => {
+              MultipleChoices(Json.obj(
+                "message" -> this.messagesApi("version.download.confirm.body.api").split('\n'),
+                "post" ->  self.confirmDownload(author, slug, target, Some(dlType.id), token).absoluteURL(),
+                "url" -> self.downloadJarById(p.pluginId, version.name, Some(token)).absoluteURL(),
+                "token" -> token)
+              )
+            })
           } else {
-            for {
-              warn <- warning
-              nonReviewed <- version.channel.map(_.isNonReviewed)
-            } yield MultipleChoices(views.unsafeDownload(data.project, version, nonReviewed , dlType, token)).withCookies(warn.cookie)
+            val userAgent = request.headers.get("User-Agent")
+            var curl: Boolean = false
```

Comment:
Do we need the types annotations for these?

Reply:
Just moved the code from above.
But its gone now.

Also we might want to remove the user agent checking entirely.
Using CLI should be api = true anyways.